pyccl/boltzmann.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+
+try:
+    import classy
+    HAVE_CLASS = True
+except ImportError:
+    HAVE_CLASS = False
+
+from . import ccllib as lib
+from .pyutils import check
+from .pk2d import Pk2D
+
+
+def get_class_pk_lin(cosmo):
+    """Run CLASS and return the linear power spectrum.
+
+    Args:
+        cosmo (:obj:`Cosmology`): Cosmological parameters.
+            The cosmological parameters with which to run CLASS.
+
+    Returns:
+        pk_lin (:obj:`Pk2D`): power spectrum object
+            The linear power spectrum.
+    """
+
+    assert HAVE_CLASS, (
+        "You must have the python wrapper for CLASS "
+        "installed to run CCL with CLASS!")
+
+    params = {
+        "output": "mPk",
+        "non linear": "none",
+        "P_k_max_1/Mpc": cosmo.cosmo.spline_params.K_MAX_SPLINE,
+        "z_max_pk": 1.0/cosmo.cosmo.spline_params.A_SPLINE_MINLOG_PK-1.0,
+        "modes": "s",
+        "lensing": "no",
+        "h": cosmo["h"],
+        "Omega_cdm": cosmo["Omega_c"],
+        "Omega_b": cosmo["Omega_b"],
+        "Omega_k": cosmo["Omega_k"],
+        "n_s": cosmo["n_s"]}
+
+    # cosmological constant?
+    # set Omega_Lambda = 0.0 if w !=-1 or wa != 0
+    if cosmo['w0'] != -1 or cosmo['wa'] != 0:
+        params["Omega_Lambda"] = 0
+        params['w0_fld'] = cosmo['w0']
+        params['wa_fld'] = cosmo['wa']
+
+    # neutrino parameters
+    # massless neutrinos
+    if cosmo["N_nu_rel"] > 1e-4:
+        params["N_ur"] = cosmo["N_nu_rel"]
+    else:
+        params["N_ur"] = 0.0
+
+    # massive neutrinos
+    if cosmo["N_nu_mass"] > 0:
+        params["N_ncdm"] = cosmo["N_nu_mass"]
+        masses = lib.parameters_get_nu_masses(cosmo._params, 3)
+        params["m_ncdm"] = ", ".join(
+            ["%g" % m for m in masses[:cosmo["N_nu_mass"]]])
+
+    params["T_cmb"] = cosmo["T_CMB"]
+
+    # if w have sigma8, we need to find A_s
+    if np.isfinite(cosmo["A_s"]):
+        params["A_s"] = cosmo["A_s"]
+    else:
+        assert np.isfinite(cosmo["sigma8"])
+        A_s_fid = 2.43e-9 * (cosmo["sigma8"] / 0.87659)**2
+        params["A_s"] = A_s_fid
+
+    model = None
+    try:
+        model = classy.Class()
+        model.set(params)
+        model.compute()
+
+        # if we are using sigma8, we rerun the code with a new A_s
+        # FIXME - I don't think rerunning class is needed...
```

Comment:
I think you're right if now we're computing our own halofit.

Reply:
Awesome! I've opened another issue. I will address this there. #651